core/src/main/java/org/apache/calcite/rel/rules/FilterCorrelateRule.java

```diff
@@ -86,8 +86,10 @@ public void onMatch(RelOptRuleCall call) {
         aboveFilters,
         JoinRelType.INNER,
         false,
-        !corr.getJoinType().toJoinType().generatesNullsOnLeft(),
-        !corr.getJoinType().toJoinType().generatesNullsOnRight(),
+        corr.getJoinType().returnsJustFirstInput()
```

Comment:
Stupid question: are we 100% sure that in the future we will never add new join types (RIGHT / FULL) to SemiJoinType enum?

Reply:
No we are not sure yet adding new types in a public enumerations affects backward compatibility so I guess it will not happen often. If at some point in the future somebody does that he will have to consider usecases like this one.

Reply:
Ok, I'll simplify then this line and the next one